core/embed/rust/src/ui/model_tt/layout.rs

```diff
@@ -4,30 +4,31 @@ use crate::{
     error::Error,
     micropython::{buffer::Buffer, obj::Obj},
     ui::{
-        component::{Child, Text},
+        component::{Child, FormattedText},
         display,
         layout::obj::LayoutObj,
     },
     util,
 };
 
 use super::{
-    component::{Button, Dialog, DialogMsg},
+    component::{Button, ButtonMsg, Dialog, DialogMsg},
     theme,
 };
 
-impl<T> TryFrom<DialogMsg<T>> for Obj
+impl<T> TryFrom<DialogMsg<T, ButtonMsg, ButtonMsg>> for Obj
 where
     Obj: TryFrom<T>,
-    Error: From<<T as TryInto<Obj>>::Error>,
+    Error: From<<Obj as TryFrom<T>>::Error>,
 {
     type Error = Error;
 
-    fn try_from(val: DialogMsg<T>) -> Result<Self, Self::Error> {
+    fn try_from(val: DialogMsg<T, ButtonMsg, ButtonMsg>) -> Result<Self, Self::Error> {
         match val {
             DialogMsg::Content(c) => Ok(c.try_into()?),
-            DialogMsg::LeftClicked => 1.try_into(),
-            DialogMsg::RightClicked => 2.try_into(),
+            DialogMsg::Left(ButtonMsg::Clicked) => 1.try_into(),
```

Comment:
Got this in todo list in #1922. I'm wondering whether the `Msg -> Obj` conversion should be defined for every Msg like here or whether this should be handled per-layout object. Either way most of the basic confirmations should probably return `Obj::const_true` and `Obj::const_false`.

Reply:
i'd prefer singletons `CONFIRMED` and `CANCELED` like we have now -- it makes it much clearer in case there's more than two possible results from a particular layout

Reply:
I think even constant strings are ok, and I'd avoid exporting these as Objs when we can.

Reply:
strings will collide if someone sets their passphrase to `CONFIRMED` :)

Reply:
Yes, hence the "when we can".